### PR TITLE
Magnet Side Door Sensor Falling Off Detection (CU-86794et47)

### PR DESCRIPTION
**Pull request by Darrickg**
This PR adds a feature to detect if the magnet side of the door has fallen off and sends an alert to the client.

Test Plan:

- [x] Existing test cases pass
- [x] New integration test cases for server side pass
- [x] Deployment to dev is successful
  - [x] Sent message at original threshold
  - [x] Sent message at follow up threshold

### CODE REVIEW

File: .travis.yml
Comment (micahdbak) [inline .travis.yml:29]: Please add a Test Plan to your PR description; it should outline that you've deployed it to dev, what you've tested while deployed to dev, and that the normal chatbot flow completes.

Take a look at Elana's PR for an example test plan: https://github.com/bravetechnologycoop/BraveButtons/pull/226

Or one of my previous test plans: https://github.com/bravetechnologycoop/BraveButtons/pull/216

File: .travis.yml
Comment (micahdbak) [inline .travis.yml:32]: Good that you added this here! (Is 432 the number of heartbeats in 5 days?)

File: CHANGELOG.md
Comment (micahdbak) [inline CHANGELOG.md:22]: Please move these lines into the [Unreleased] header of the CHANGELOG.md. The other changes are from the released version.

It should be like this:

```
# [Unreleased]

## Added

- (your changes)

# [10.9.0] - XXXX-XX-XX

...
```

File: firmware/boron-ins-fsm/src/imDoorSensor.cpp
Comment (micahdbak) [inline firmware/boron-ins-fsm/src/imDoorSensor.cpp:85]: good!

File: firmware/boron-ins-fsm/src/imDoorSensor.cpp
Comment (micahdbak) [inline firmware/boron-ins-fsm/src/imDoorSensor.cpp:92]: yippee!!

File: firmware/boron-ins-fsm/src/imDoorSensor.cpp
Comment (micahdbak) [inline firmware/boron-ins-fsm/src/imDoorSensor.cpp:248]: very nice exporting

Comment (micahdbak) [inline firmware/boron-ins-fsm/src/imDoorSensor.cpp:248]: Question: if you're using consecutiveDoorOpenHeartbeatCount in src/stateMachine.cpp, then is there a reason you have this function in src/imDoorSensor.cpp? It may be redundant if the variable is global

Comment (micahdbak) [reply firmware/boron-ins-fsm/src/imDoorSensor.cpp:248]: Otherwise, please call this function in src/stateMachine.cpp and don't have the extern mention in src/imDoorSensor.h to ensure that the variable is only modified in src/imDoorSensor.cpp

File: server/resources/translations/common.en.json
Comment (micahdbak) [inline server/resources/translations/common.en.json:21]: Please add your translations here for English-FrenchCA translations: https://docs.google.com/spreadsheets/d/1Nna-TF9kwtB3xYhZl2GHrYbExVaYc80P/edit#gid=304727870

And for English-Spanish: https://docs.google.com/spreadsheets/d/1oSWm6Jm0A88X7hrYmH8fjoc-gqpPV9JF/edit#gid=1656917058

Use Google Translate for translating from English, but ping Oona in #sensor-aa-general in Slack and let her know you've added new messages. Also paste your English version there, and other people in the team can give some feedback on the wording

**Review by micahdbak — COMMENTED** [review]
Looks good so far, please take a look at my comments. I'm going to look at your test cases after the devices meeting

File: .travis.yml
Comment (Darrickg) [reply .travis.yml:32]: 432 is an arbitrary number of days in heartbeats (not 5). Was only using it for test cases, but we should definitely decide on the actual number of days on when to send the alert.

File: server/resources/translations/common.en.json
Comment (Darrickg) [reply server/resources/translations/common.en.json:21]: added and sent!

File: CHANGELOG.md
Comment (Darrickg) [reply CHANGELOG.md:22]: fixed in latest push

File: firmware/boron-ins-fsm/src/imDoorSensor.cpp
Comment (Darrickg) [reply firmware/boron-ins-fsm/src/imDoorSensor.cpp:248]: this is a leftover function. it's not used at all. removed in latest push

File: server/vitals.js
Comment (Darrickg) [reply server/vitals.js:311]: made everything OpenDoor to be consistent. also made the helpers variables in latest push

File: server/test/integration/vitalsTest/handleHeartbeatTest.js
Comment (Darrickg) [reply server/test/integration/vitalsTest/handleHeartbeatTest.js:474]: removed to.have.been.calledWith in latest push

File: .travis.yml
Comment (Darrickg) [reply .travis.yml:29]: done

**Comment by Darrickg** [comment]
pushed new changes! lmk if there is anything else you want me to change!

**Comment by trisapeace** [comment]
Task linked: [CU-86794et47 Sentry Alert for when the magnet side of the door sensor falls off](https://app.clickup.com/t/86794et47)

File: .travis.yml
Comment (micahdbak) [reply .travis.yml:32]: Kewl

File: server/vitals.js
Comment (micahdbak) [reply server/vitals.js:311]: Yayyy I like this version! Thanks!

**Review by micahdbak — APPROVED** [review]
Looks good, am happy with these changes being merged. Well done! Make sure to Squash & Merge.

**Comment by Darrickg** [comment]
@micahdbak should i merge now, even though the translations are possibly off and the variables for the threshold are not discussed yet (they are still arbitrary days that holds no actual meaning or thought)?

**Comment by Darrickg** [comment]
everything works now, just waiting for @johnny-brave to look everything over before merging

File: CHANGELOG.md
Comment (johnny-brave) [inline CHANGELOG.md:17]: function name should be a bit more descriptive. It's not guaranteed that the door sensor has fallen off. The alert just indicates that there has been no door closing activity. The most likely cause is that the magnet side of the door has been removed. I would suggest changing `sendFallOffAlert` to `sendDoorInactivityAlert`

File: server/test/integration/vitalsTest/handleHeartbeatTest.js
Comment (johnny-brave) [inline server/test/integration/vitalsTest/handleHeartbeatTest.js:99]: similarly, I think all instances of "doorfallOff" should be reworded to "doorInactivty"

File: server/resources/translations/common.en.json
Comment (johnny-brave) [inline server/resources/translations/common.en.json:21]: I wrote my suggestions for the message wording in the sensor-aa-general slack channel. I also suggest changing `sensorFallOff` to 'sensorInactivty'

**Review by johnny-brave — COMMENTED** [review]
Good work! The functionality looks solid. The only things I would change are some variable naming

File: firmware/boron-ins-fsm/src/imDoorSensor.cpp
Comment (Darrickg) [reply firmware/boron-ins-fsm/src/imDoorSensor.cpp:85]: fixed in latest push!

File: server/test/integration/vitalsTest/handleHeartbeatTest.js
Comment (Darrickg) [reply server/test/integration/vitalsTest/handleHeartbeatTest.js:99]: fixed in latest push!

File: server/resources/translations/common.en.json
Comment (Darrickg) [reply server/resources/translations/common.en.json:21]: fixed in latest push!

File: CHANGELOG.md
Comment (Darrickg) [reply CHANGELOG.md]: fixed in latest push!

**Comment by Darrickg** [comment]
Everything works, just waiting for Johnny to give me final approval before I merge!

**Review by johnny-brave — APPROVED** [review]
Looks great! Thanks for making all the variable names consistent. This is ready to Squash and Merge

**Pull request merged**
